cluster-agent/controllers/managed-gitops/eventloop/operation_event_loop.go

```diff
@@ -541,9 +548,16 @@ func processOperation_SyncOperation(ctx context.Context, dbOperation db.Operatio
 // returns shouldRetry, error
 func terminateExistingOperation(ctx context.Context, dbApplication *db.Application, opConfig operationConfig) (bool, error) {
 
-	if err := utils.TerminateOperation(ctx, dbApplication.Name, opConfig.argoCDNamespace, opConfig.credentialService,
+	if err := opConfig.syncService.terminateOperation(ctx, dbApplication.Name, opConfig.argoCDNamespace, opConfig.credentialService,
 		opConfig.eventClient, time.Duration(5*time.Minute), opConfig.log); err != nil {
 
+		// we can return if there are no sync operations in progress
+		terminatedOperationErr := "Unable to terminate operation. No operation is in progress"
```

Comment:
Hardcoding a non-constant error message from Argo CD makes me a bit nervous. I think we should investigate alternatives: can you see if Argo CD exposes an API that would allow us to tell if there was an operation in progress? That way we we could use it to avoid this error message altogether. :thinking: 

Something like
```go
func terminateExistingOperation(ctx context.Context, dbApplication *db.Application, opConfig operationConfig) (bool, error) {
   if (operation is not running on application) { return false, nil }
   // (... rest of the code ...)
}
```

The API I'm looking for would be something like an `isOperationRunning()` or `getCurrentOperation()` type function that we can call on Argo CD to check if an operation is running, and/or determine if terminating the operation is valid. 

If so, we can open a new story to switch to this (or handle it on this story/PR if you think the work is small enough: up to you). 

Also open to alternative proposals to fix, if you have suggestions. In any case, let me know what you think!